ui/app/AppLayouts/stores/CloneModel.qml

```diff
@@ -0,0 +1,55 @@
+import QtQuick 2.15
+
+/// Helper item to clone a model and alter its data without affecting the original model
+/// \beware this is not a proxy model. It clones the initial state
+///     and every time the instance changes and doesn't adapt when the data
+///     in the source model \c allNetworksModel changes
+/// \beware use it with small models and in temporary views (e.g. popups)
+/// \note requires `rowData` to be implemented in the model
+/// \note tried to use SortFilterProxyModel with but it complicates implementation too much
+ListModel {
+    id: root
+
+    required property var sourceModel
+
+    /// Roles to clone
+    required property var roles
+
+    /// Roles to override or add of the form { role: "roleName", transform: function(modelData) { return newValue } }
+    property var rolesOverride: []
+
+    Component.onCompleted: cloneModel(sourceModel)
+    onSourceModelChanged: cloneModel(sourceModel)
+
+    function rowData(index, roleName) {
+        return get(index)[roleName]
+    }
+
+    function findIndexForRole(roleName, value) {
+        for (let i = 0; i < count; i++) {
+            if(get(i)[roleName] === value) {
+                return i
+            }
+        }
+        return -1
+    }
+
+    function cloneModel(model) {
```

Comment:
@stefandunca I'm bit late bc it's already merged, but please take a look at `status-desktop/ui/StatusQ/src/StatusQ/Core/Utils/ModelUtils.qml`. There are some primitives doing basically the same job. E.g. method `modelToArray` could be easily used for cloning. In the simplest form it's `append(ModelUtils.modelToArray(model, roles)`. The additional benefit is the fact that `modelToArray` handles properly all `QAbstractItemModel`, not only ListModel's.

I also think that it could be beneficial to move `CloneModel.qml` to e.g. `status-desktop/ui/StatusQ/src/StatusQ/Core/Utils/` bc it seems to be a generic utility.

Reply:
It is a nice surprise to find out about these utilities. Thanks for the heads up. Next time, I will use them along with the C++ capabilities instead of JS. F.Y.I. @IvanBelyakoff